pulsar-common/src/main/java/org/apache/pulsar/common/util/TrustManagerProxy.java

```diff
@@ -74,11 +76,16 @@ private void updateTrustManager() throws CertificateException, KeyStoreException
             FileNotFoundException, IOException {
         CertificateFactory factory = CertificateFactory.getInstance("X.509");
         try (InputStream inputStream = new FileInputStream(certFile.getFileName())) {
-            X509Certificate certificate = (X509Certificate) factory.generateCertificate(inputStream);
-            String alias = certificate.getSubjectX500Principal().getName();
-            KeyStore keyStore = KeyStore.getInstance("JKS");
+            KeyStore keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
             keyStore.load(null);
-            keyStore.setCertificateEntry(alias, certificate);
+
+            Collection<? extends Certificate> certificates = factory.generateCertificates(inputStream);
```

Comment:
Did you notice the `SecurityUtility#loadCertificatesFromPemFile` method? It looks like that could replace some of this logic. Also, the logic in this method appears to duplicate some of the logic in the file you're modifying: https://github.com/apache/pulsar/blob/ec3821176612621e24dfbc4345525849a729fb06/pulsar-common/src/main/java/org/apache/pulsar/common/util/SecurityUtility.java#L325-L331.